include/verilated_profiler.h

```diff
@@ -23,58 +23,202 @@
 #define VERILATOR_VERILATED_PROFILER_H_
 
 #include "verilatedos.h"
-#include "verilated.h"  // for VerilatedMutex and clang annotations
 
-#include <deque>
+#ifndef VL_PROFILER
+#error "verilated_profiler.h/cpp expects VL_PROFILER (from --prof-{exec, pgo}"
+#endif
+
+#include "verilated.h"
+
+#include <array>
+#include <atomic>
+#include <cassert>
 #include <string>
+#include <type_traits>
+#include <vector>
+
+class VlExecutionProfiler;
+
+//=============================================================================
+// Macros to simplify generated code
+
+#define VL_ADD_EXECUTION_TRACE_RECORD(vlSymsp) \
+    if (VL_UNLIKELY((vlSymsp)->__Vm_executionProfiler.enabled())) \
+    (vlSymsp)->__Vm_executionProfiler.addRecord()
```

Comment:
```suggestion
#define VL_EXEC_TRACE_ADD_RECORD(vlSymsp) \
    do { if (VL_UNLIKELY((vlSymsp)->__Vm_executionProfiler.enabled())) \
    (vlSymsp)->__Vm_executionProfiler.addRecord() } while(0)
```

(And replace elsewhere. Name to match that we use "exec" instead of "execution" everywhere else. Also add do/while wrapper to make sure is safer to use)

Reply:
This is used as ` VL_EXEC_TRACE_ADD_RECORD(vlSymsp).evalBegin()` etc, so the `do` `while` is not good. It's however only in generated code, so we can just get code generation right. Will fix the name.